src/main/scala/com/sky/kafka/topicloader/config/topicLoaderConfig.scala

```diff
@@ -0,0 +1,71 @@
+package com.sky.kafka.topicloader.config
+
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+
+import cats.data.Validated
+import cats.implicits._
+import com.typesafe.config.{Config => TypesafeConfig, ConfigException}
+
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Try
+
+trait FromConfig[A] {
+  def fromConfig(path: String, config: TypesafeConfig): A
```

Comment:
the idiomatic Scala way to do this would be:
`def fromConfig(path: String, config: TypesafeConfig): Either[ConfigError, A]`
instead of throwing exceptions in the implementations.

The goal should be to not throw exceptions anywhere except in one place, the `loadOrThrow` method which just calls fromConfig and then throws if the result is a left

Reply:
I think the problem here is that the `TypeSafe` config already throws a `RuntimException` if it can't load for some reason. So if we change from `Try`s to `Either`s, our `loadOrThrow` won't aggregate the `TypeSafe` errors, only throw the first one it encounters. It will aggregate all of our custom errors though, only if there's no TS errors. In that case I should probably remove the test that tests the TypeSafe error, the int parsing of 999999999999

Reply:
we can use a validated nel to accumulate errors.

i'm also wondering if the type class is even worth it, considering this library is only ever going to load one type of config class `TopicLoaderConfig`. At which point we ditch the type class and just have a function:

```scala
  def load(config: TypesafeConfig): ValidatedNec[ConfigError, TopicLoaderConfig]

  def loadOrThrow(config: TypesafeConfig) = load(config) match {// throw if invalid else return topicLoaderConfig}
```

Reply:
yeah this is good, nice one